airflow/sensors/base.py

```diff
@@ -281,6 +282,16 @@ def run_duration() -> float:
         self.log.info("Success criteria met. Exiting.")
         return xcom_value
 
+    def resume_execution(self, next_method: str, next_kwargs: dict[str, Any] | None, context: Context):
+        try:
+            return super().resume_execution(next_method, next_kwargs, context)
+        except (AirflowException, TaskDeferralError) as e:
+            if self.soft_fail:
+                raise AirflowSkipException(e)
+            raise
+        except Exception:
```

Comment:
Should we raise `AirflowSkipException` when `sefl.soft_fail` is `True` here?

Reply:
IMHO, I don't think so, soft fail should be used when the sensor returns a False value or it's timeout, so we cannot skip the task if there is a failure. I'm preparing a PR for that case.